src/jquery.telemetry.js

```diff
@@ -325,52 +325,59 @@ $.widget("telemetry.histogramfilter", {
     return prettyNames.hasOwnProperty(filterName) ? prettyNames[filterName] : "Any " + filterName;
   },
   
-  _windowsVersionNames: {"5.0": "2000", "5.1": "XP", "5.2": "XP Pro x64", "6.0": "Vista", "6.1": "7", "6.2": "8", "6.3": "8.1", "10.0": "10"},
-  _getOptions: function histogramfilter__getOptions(filterName, options) {
+  _systemNames: {"WINNT": "Windows", "Windows_95": "Windows 95", "Darwin": "OS X"},
+  _windowsVersionNames: {"5.0": "2000", "5.1": "XP", "5.2": "XP Pro x64", "6.0": "Vista", "6.1": "7", "6.2": "8", "6.3": "8.1", "6.4": "10 (Tech Preview)", "10.0": "10"},
+  _windowsVersionOrder: {"5.0": 0, "5.1": 1, "5.2": 2, "6.0": 3, "6.1": 4, "6.2": 5, "6.3": 6, "6.4": 7, "10.0": 8},
+  _darwinVersionPrefixes: {
+    "1.2.": "Kodiak", "1.3.": "Cheetah", "1.4.": "Puma", "6.": "Jaguar",
+    "7.": "Panther", "8.": "Tiger", "9.": "Leopard", "10.": "Snow Leopard",
+    "11.": "Lion", "12.": "Mountain Lion", "13.": "Mavericks", "14.": "Yosemite",
+  },
+  _archNames: {"x86": "32-bit", "x86-64": "64-bit"},
+  _getHumanReadableOptions: function histogramfilter__getHumanReadableOptions(filterName, options) {
     if (filterName === "OS") {
       // Replace OS names with pretty OS names where possible
-      var systemNames = {"WINNT": "Windows", "Darwin": "OS X"};
+      var systemNames = this._systemNames;
       return options.map(function(option) {
         return systemNames.hasOwnProperty(option) ? systemNames[option] : option;
       });
     } else if (filterName === "osVersion") {
-      // look for the selected OS filter in order to determine which pretty OS names to use
-      // or give up if we can't find it
-      var system = null;
-      var realValue = this._getOptionRealValue;
-      this._filterList.forEach(function(filter){
-        if (filter.histogram.filterName() == "OS") {
-          system = (typeof filter.select.val === "function") ?
-                   filter.select.val() : filter.select;
-          system = realValue("OS", system);
-        }
-      });
-      if (system === null) {
-        return options;
-      }
+      // get the currently selected OS
+      var system = $("#histogram-filters input.filter-OS").val() || this._getStarName("OS");
+      system = this._getHumanReadableOptionRealValue("OS", system);
       
       if (system === "WINNT") {
         var versionNames = this._windowsVersionNames;
-        return options.map(function(option) {
+        var versionOrder = this._windowsVersionOrder;
+        var starName = this._getStarName(filterName);
+        return options.sort(function(a, b) {
```

Comment:
Right, `_getHumanReadableOptions` is called whenever a select needs to be changed, and this particular line is only executed when the OS selector is set to "Windows".